pkg/shardddl/optimism/keeper.go

```diff
@@ -46,8 +50,13 @@ func (lk *LockKeeper) TrySync(info Info, tts []TargetTable) (string, []string, e
 	defer lk.mu.Unlock()
 
 	if l, ok = lk.locks[lockID]; !ok {
-		lk.locks[lockID] = NewLock(lockID, info.Task, info.DownSchema, info.DownTable, info.TableInfoBefore, tts)
+		lk.locks[lockID] = NewLock(cli, lockID, info.Task, info.DownSchema, info.DownTable, info.TableInfoBefore, tts)
 		l = lk.locks[lockID]
+		if lk.colm != nil {
```

Comment:
could we move this "recovering columns" to `recoverLocks`? We could iterate every locks and check `colm`. It could also remove `SetColumnMap`

Reply:
@lichunzhu here's a comment waiting to address